libs/utils/lib/path.ts

```diff
@@ -16,23 +16,37 @@ import { HFile, HPath } from "@howdypix/shared-types";
  *
  */
 
-export function hjoin(howdyfile: HFile) {
-  return `@${howdyfile.source}:` + path.join(howdyfile.dir, howdyfile.file);
+export function hjoin(howdyfile: HFile): HPath {
+  return (
+    `@${howdyfile.source}:` +
+    path.join(howdyfile.dir ?? "", howdyfile.file ?? "")
+  );
 }
 
 export function hparse(hpath: HPath): HFile {
```

Comment:
Yes! I had some issue that I needed to fix like if we are at a root dir, then we have `@main:.`, because the dir is `.`... 
We should also URL encode it just in case...